altair/utils/core.py

```diff
@@ -227,9 +227,13 @@ def parse_shorthand(shorthand, data=None):
     {'aggregate': 'count', 'type': 'quantitative'}
     """
     attrs = _parse_shorthand(shorthand)
-    if isinstance(data, pd.DataFrame) and 'type' not in attrs:
+    if isinstance(data, pd.DataFrame):
         if 'field' in attrs and attrs['field'] in data.columns:
-            attrs['type'] = infer_vegalite_type(data[attrs['field']])
+                if 'type' not in attrs:
+                    attrs['type'] = infer_vegalite_type(data[attrs['field']])
+                if hasattr(data,'__geo_interface__'): #TODO: Add descripion in shorthand
+                    attrs['field']='properties.'+attrs['field']
+
```

Comment:
Something strange is going on here: is not checking for ``type`` in ``attrs`` necessary for this PR? And why is this block doubly indented?

Reply:
`'type' not in attrs` this is still checked, but in the line below (232).
The idea behind: to allow user work with GeoDataFrame like as regular DataFrame but showing geoshapes attached to its rows. 
**Implementation details:**
First of all GeoDataFrame is subclass of `pd.DataFrame` with `__geo_interface__`. So it is instance of `pd.DataFrame` that has attribute `__geo_interface__`.  GeoPandas are stored as geojson FeatureCollection with each row as Feature object where all columns are placed in properties object. Sample will be more informative:
```javascript
{ /* vega-light data*/
    "format": {
        "property": "features", /* generate geoshape for each row*/
        "type": "json"
    },
    "values": { /* valid geojson for all rows*/
        "type": "FeatureCollection",
        "features": [
            { /* valid geojson for each row*/
                "type": "Feature",
                "properties": { /* column values  */
                    "pop_est": 12799293.0,
                    "continent": "Africa",
                },
                "geometry": { /* geometry of the row  */
                    "type": "MultiPolygon",
                    "coordinates": [ /* a lot of numbers*/]
                }
            }
        ]
    }
}
```
So first step is to add ["property": "features"] to vega-light data format description. That splits valid geojson stored in values back to rows of GeoDataFrame (it is possible to replace this step with storing content of "features" directly into "values" but that will made "values" invalid geojson).

Next is access to column values of GeoDataFrame. Values are accessible from chart as "properties.column_name". I hoped to simplify user experience by adding "properties." in shorthand. Now if user use shorthands he will get same behavior for GeoDataFrame as for DataFrame (take a look to updated description of PR)

Reply:
indentation fixed

Reply:
Not needed anymore due new save format